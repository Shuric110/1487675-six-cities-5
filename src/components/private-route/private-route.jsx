import React from "react";
import PropTypes from "prop-types";
import {Route, Redirect} from "react-router-dom";
import {connect} from "react-redux";
import {AuthorizationStatus, AppRoute} from "../../const";


const PrivateRoute = (props) => {
  const {path, exact, authorizationStatus} = props;
  const render = props.render || (() => props.children);

  return (
    <Route
      path={path}
      exact={exact}
      render={(routeProps) => {
        return (
          authorizationStatus === AuthorizationStatus.AUTH
            ? render(routeProps)
            : <Redirect to={{
              pathname: AppRoute.LOGIN,
              state: {returnUrl: routeProps.location.pathname}
            }} />
        );
      }}
    />
  );
};

PrivateRoute.propTypes = {
  authorizationStatus: PropTypes.string.isRequired,
  path: PropTypes.string.isRequired,
  exact: PropTypes.bool,
  render: PropTypes.func,
  children: PropTypes.node
};

const mapStateToProps = (state) => ({
  authorizationStatus: state.USER.authorizationStatus,
});


export {PrivateRoute};
export default connect(mapStateToProps)(PrivateRoute);
