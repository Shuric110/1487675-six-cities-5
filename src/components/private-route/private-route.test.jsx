import React from "react";
import renderer from "react-test-renderer";
import {MemoryRouter as Router} from 'react-router-dom';
import {PrivateRoute} from "./private-route";

describe(`Render PrivateRoute`, () => {
  it(`PrivateRoute for authenticated user with children renders correctly`, () => {
    const tree = renderer
      .create(
          <Router>
            <PrivateRoute
              authorizationStatus="AUTH"
              path="/"
            >
              Inner content in children
            </PrivateRoute>
          </Router>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`PrivateRoute for authenticated user with render prop renders correctly`, () => {
    const tree = renderer
      .create(
          <Router>
            <PrivateRoute
              authorizationStatus="AUTH"
              path="/"
              render={() => <React.Fragment>Inner content in render prop</React.Fragment>}
            />
          </Router>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`PrivateRoute for unauthenticated user renders correctly`, () => {
    const tree = renderer
      .create(
          <Router>
            <PrivateRoute
              authorizationStatus="NO_AUTH"
              path="/"
            >
              Inner content
            </PrivateRoute>
          </Router>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
