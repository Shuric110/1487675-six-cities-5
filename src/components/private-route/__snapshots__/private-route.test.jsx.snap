// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render PrivateRoute PrivateRoute for authenticated user with children renders correctly 1`] = `"Inner content in children"`;

exports[`Render PrivateRoute PrivateRoute for authenticated user with render prop renders correctly 1`] = `"Inner content in render prop"`;

exports[`Render PrivateRoute PrivateRoute for unauthenticated user renders correctly 1`] = `null`;
